include/engine/plugins/match.hpp

```diff
@@ -185,55 +187,37 @@ template <class DataFacadeT> class MapMatchingPlugin : public BasePlugin
             subtrace.values["confidence"] = sub.confidence;
         }
 
-        JSONDescriptor<DataFacadeT> json_descriptor(facade);
-        json_descriptor.SetConfig(route_parameters);
+        osrm::engine::guidance::ApiResponseGenerator<DataFacadeT> response_generator(facade);
```

Comment:
small nitpicking, but we could create type inference `make*` helper functions, so that we can just say

    auto response_generator = makeApiResponseGenerator(facade);